### PR TITLE
Removed incorrect check for returncode

### PR DESCRIPTION
**Pull request by ofayans**
The server installation in most cases returns response code 0 no matter what
happens except for really severe errors. In this case when we try to uninstall
the middle replica of a line topology, it fails, notifies us that we should use
'--ignore-topology-disconnect', but returns 0

https://fedorahosted.org/freeipa/ticket/3230

### CODE REVIEW

File: ipatests/test_integration/tasks.py [hunk 1]
```diff
@@ -1190,7 +1190,7 @@ def run_server_del(host, server_to_delete, force=False,
 def assert_error(result, stderr_text, returncode=None):
     "Assert that `result` command failed and its stderr contains `stderr_text`"
     assert stderr_text in result.stderr_text, result.stderr_text
-    if returncode:
+    if returncode is not None:
```
Comment (mbasti-rh) [inline ipatests/test_integration/tasks.py]: Why this change is needed?

Comment (mbasti-rh) [inline ipatests/test_integration/tasks.py]: Got it

**Comment by mbasti-rh** [comment]
ACK: Removed incorrect check for returncode
NACK:  Several fixes in replica_promotion tests -- missing explanation in commit messages why this change is needed, if changes are unrelated should be in multiple patches 
ACK: Changed addressing to the client hosts to be replicas
ACK: Xfailed the tests due to a known bug with replica preparation

**Comment by ofayans** [comment]
The commit message was updated

**Comment by mbasti-rh** [comment]
I went through commit messages again and I changed my mind, NACK:

1)
commit: Removed incorrect check for returncode

This commit contains incorrect ticket, issue has not been fixed by this commit

2)
commit: Several fixes in replica_promotion tests

This does not have ticket, is okay to push it just to master?

3)
All commits have different tickets, so I have to manually split hashes and put it to proper tickets. Leave it as is for this PR to avoid mess, but in future create one PR for one ticket

**Comment by ofayans** [comment]
@mbasti-rh,
1. Fixed
2. It's OK, but we won't have working tests in 4.3 branch. Should I create a ticket?
3. Gotya :)

**Comment by mbasti-rh** [comment]
Ad 2. yes

**Comment by ofayans** [comment]
Done

**Comment by mbasti-rh** [comment]
**Removed incorrect check for returncode**

Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/22b0e8a9eb9eb3d47131c6784d70dd409d5b889b
ipa-4-4:
https://fedorahosted.org/freeipa/changeset/e265853d055caf7e3d17316eee6e25aa26bbf2a9

**Comment by mbasti-rh** [comment]
**Several fixes in replica_promotion tests**

Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/39c15ecdcdcbc2a0b651b0f89080789dd806e998
ipa-4-4:
https://fedorahosted.org/freeipa/changeset/cd6adafbf699da48ab877e77ac9c1cc1dd26bf61

**Comment by mbasti-rh** [comment]
**Changed addressing to the client hosts to be replicas**

Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/ac78d191ded6721cf1051413cdd6862c0362e66b
ipa-4-4:
https://fedorahosted.org/freeipa/changeset/de4a1fc0df5474f268c7ed08ffb802110631c13f
ipa-4-3:
https://fedorahosted.org/freeipa/changeset/8ed4a4ba6392deb7972ddc07593d649926065c72

**Comment by mbasti-rh** [comment]
**Xfailed the tests due to a known bug with replica preparation**

Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/1e484d010b653b8ac06425ca602ba5c9e950ed89
